### PR TITLE
LPS-193689 - Create Modal to Bind an Object to a Root Object

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://liferay.atlassian.net/browse/LPS-193689

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -15182,6 +15187,7 @@ select-the-publication-to-move-the-selected-changes-to=Select the publication to
 select-the-publication-x-to-review-the-change=Select the publication {0} to review the change.
 select-the-ratings-type-for-the-following-applications=Select the ratings type for the following applications.
 select-the-reassignment-type=Select the reassignment type.
+select-the-root-object-path=Select the Root Object Path
```
Comment (naraandrade) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:15190]: @danvb112 Maybe BChan will consider this as a sentence, but let's wait to see :)

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ModalBindToRootObject.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+	Input,
+	SingleSelect,
+	getLocalizableLabel,
+	openToast,
+} from '@liferay/object-js-components-web';
+import {createResourceURL, fetch} from 'frontend-js-web';
+
+interface ModalBindToRootObjectProps {
+	baseResourceURL: string;
+	onvisibilityChange: () => void;
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ModalBindToRootObject.tsx:24]: @danvb112 Just a small change, please capitalize the 'v' on `onVisibilityChange`.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ModalBindToRootObject.tsx [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+	};
+
+	const handleSelectRootObject = (
+		option: RootObjectPathOption,
+		depth: number
+	) => {
+		setLoading(true);
+
+		const newSelectedRootObjects = selectedRootObjects;
+		newSelectedRootObjects[depth] = option;
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ModalBindToRootObject.tsx:118]: @danvb112 Please, add a blank line between the creation and usage of the const.

**Review by naraandrade — CHANGES_REQUESTED** [review]
@danvb112 Great job with this modal! You should link this in your Level Positioning as an example of a complex feature you have worked with. Just left a few comments.

**Comment by danvb112** [comment]
ci:test:sf

**Review by naraandrade — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/2508

**Pull request closed without merging**